stix_shifter/src/modules/qradar/aql_query_constructor.py

```diff
@@ -217,8 +217,13 @@ def translate_pattern(pattern: Pattern, data_model_mapping, result_limit, timera
     select_statement = translated_where_statements.dmm.map_selections()
     queries = []
     for where_statement in translated_where_statements.queries:
-        if(_test_for_start_stop(where_statement)):
-            queries.append("SELECT {} FROM events WHERE {}".format(select_statement, where_statement))
+        has_start_stop = _test_for_start_stop(where_statement)
+        if(has_start_stop):
+            query = "SELECT {} FROM events WHERE {}".format(select_statement, where_statement)
         else:
-            queries.append("SELECT {} FROM events WHERE {} {}".format(select_statement, where_statement, result_limit))
+            query = "SELECT {} FROM events WHERE {} limit {}".format(select_statement, where_statement, result_limit)
+        if(not has_start_stop):
```

Comment:
couldn't this be added to the above else statement?

Reply:
Yes, I don't know what I was thinking.